state/state.go

```diff
@@ -772,12 +805,14 @@ func (s *State) fetchMember(gID discord.GuildID, uID discord.UserID) (m *discord
 // tracksMessage reports whether the state would track the passed message and
 // messages from the same channel.
 func (s *State) tracksMessage(m *discord.Message) bool {
-	return (m.GuildID.IsValid() && s.Gateway.HasIntents(gateway.IntentGuildMessages)) ||
-		(!m.GuildID.IsValid() && s.Gateway.HasIntents(gateway.IntentDirectMessages))
+	return (m.GuildID.IsValid() &&
+		s.ShardManager.FromGuildID(m.GuildID).HasIntents(gateway.IntentGuildMessages)) ||
+		(!m.GuildID.IsValid() &&
+			s.ShardManager.FromGuildID(m.GuildID).HasIntents(gateway.IntentDirectMessages))
 }
 
 // tracksChannel reports whether the state would track the passed channel.
 func (s *State) tracksChannel(c *discord.Channel) bool {
-	return (c.GuildID.IsValid() && s.Gateway.HasIntents(gateway.IntentGuilds)) ||
-		!c.GuildID.IsValid()
+	return (c.GuildID.IsValid() && s.ShardManager.FromGuildID(c.GuildID).
+		HasIntents(gateway.IntentGuilds)) || !c.GuildID.IsValid()
```

Comment:
Weird indentation.

Reply:
This is the indention gofmt uses, what should I change? I could put the function call on the same line, but that would require an extra line.

Reply:
I personally think another line doesn't hurt and will aid in readability instead.